PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Purchases.swift
+//
+//  Created by Joshua Liebowitz on 8/18/21.
+
+// swiftlint:disable file_length function_parameter_count type_body_length
+import Foundation
+import StoreKit
+
+// MARK: Block definitions
+// NOTE: Any changes here must be reflected in PurchasesCoreSwift.h for ObjC compatibility.
+/**
+ Completion block for calls that send back a `PurchaserInfo`
+ */
+public typealias ReceivePurchaserInfoBlock = (PurchaserInfo?, Error?) -> Void
```

Comment:
🤔 perhaps we could move these to a separate file, and have an obj-c equivalent, so it's easier to keep them 1-1?

Reply:
It'll take some fiddling around to get these to play nice along different package managers (particularly SPM), but I think it should be doable

Reply:
Yeah, not a bad idea, can do a follow up. #761 761